api/docs/CMake_rundoxygen.cmake

```diff
@@ -286,7 +291,31 @@ if (embeddable)
         message(FATAL_ERROR "Cannot find menu entry for page \"Extension API\"")
       endif ()
       string(REGEX MATCH "\n[^\n]+\"Extension API\"[^\n]+\n" ext_entry "${string}")
-      string(REPLACE "${ext_entry}" "\n" string "${string}")
+      # If ext_entry ends with a [, the menu items are expanded inline and the
+      # whole Extension API menu can be moved within this file.
+      string(REGEX MATCH "\\[\n" ext_multiline "${ext_entry}")
+      if (ext_multiline)
+        # Extract everything up to but not including the newly-inserted API Reference.
+        string(REGEX REPLACE
+               "(\n[^\n]+\"Extension API\",.+)(\n\\[ \"API Reference\")"
+               "\\2" string "${string}")
+        # Remember the full Extension API match.
+        set(ext_entry ${CMAKE_MATCH_1})
+        if (NOT ext_entry)
+          message(FATAL_ERROR "Cannot move menu entry for \"Extension API\"")
+        endif ()
+        # Now insert the Extension API menu right above Disassembly Library.
+        if (NOT string MATCHES "Disassembly Library")
+          message(FATAL_ERROR "Cannot find menu entry for page \"Disassembly Library\"")
+        endif ()
+        string(REGEX REPLACE "(\n[^\n]+\"Disassembly Library)" "${ext_entry}\\1"
+          string "${string}")
+        # Set the user_docs flag since we found its contents here.
+        set(found_user_docs ON)
```

Comment:
Add a sanity check !found_user_docs down on line 349 where it finds page_user_docs.js?

Reply:
Done.